pkg/snapshotter/snapshotter_csi.go

```diff
@@ -601,7 +601,26 @@ func (c *csiDriver) RestoreVolumeClaim(opts ...Option) (*v1.PersistentVolumeClai
 				logrus.Debugf("setting size of pvc %s/%s same as snapshot size %s", pvc.Namespace, pvc.Name, quantity.String())
 				pvc.Spec.Resources.Requests[v1.ResourceStorage] = quantity
 			}
-
+		}
+		// PB-8482 if restore size is nil then set the pvc size to annotation restoreSize
+		if snapshot.Status.RestoreSize == nil {
+			restoreSizeStr, ok := snapshot.Annotations["restoreSize"]
+			if !ok {
+				return nil, fmt.Errorf("restoreSize annotation not found")
+			}
+			restoreSize, err := resource.ParseQuantity(restoreSizeStr)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse restoreSize: %v", err)
```

Comment:
nit: failed to parse restoreSize quantity provided in annotation px-backup/restoreSize